src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -383,9 +383,29 @@ CHIP_ERROR Browse(void * context, DnssdBrowseCallback callback, uint32_t interfa
     std::string regtype(type);
     std::string subtypeDelimiter = "._sub.";
     size_t position              = regtype.find(subtypeDelimiter);
+    // if we have more in the string (a subtype) then append it to the end
     if (position != std::string::npos)
     {
-        regtype = regtype.substr(position + subtypeDelimiter.size()) + "," + regtype.substr(0, position);
+        // DNSServiceBrowse allows searching by type, and by subtype (PTR records).
+        //  (ex. _matterc._udp.local. or _CM._sub._matterc._udp.local.)
+        //
+        // It does not include support for searching by instance name (SRV records).
+        //  (ex. DD200C20D25AE5F7._matterc._udp.local.)
```

Comment:
But in that string, there is no "._sub.", so we won't even be in this branch.  Why is that string relevant to this code?

Reply:
looking deeper into this code, it turns out that the instance name filter was not being handled correctly by the resolver since it was being treated as a subtype which it is not. When we know the instance, we just need to perform a resolve and not a browse.